lib/ruby_lsp/requests/formatting.rb

```diff
@@ -25,11 +25,19 @@ class Error < StandardError; end
 
       extend T::Sig
 
-      sig { params(uri: String, document: Document).void }
-      def initialize(uri, document)
+      sig { params(uri: String, document: Document, formatter: String).void }
+      def initialize(uri, document, formatter: "auto")
         super(document)
 
         @uri = uri
+        @formatter = T.let(
+          if formatter == "auto"
+            defined?(Support::RuboCopFormattingRunner) ? "rubocop" : "syntax_tree"
+          else
```

Comment:
Should we check if the formatter is supported / error if it isn't? 

In the case of VSCode we will be able to set that the variable must be `auto|rubocop|syntax_tree`, but for cases outside of that, we don't explicitly support passing the wrong type.